src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -927,6 +927,9 @@ private void parsePublicKey(final List<Credential> credentials, final XMLStreamR
         private void parsePassword(final List<Credential> credentials, final XMLStreamReader streamReader) throws XMLStreamException, RealmUnavailableException {
             parseCredential(streamReader, (algorithm, format, text) -> {
                 try {
+                    if (algorithm == null) {
+                        throw ElytronMessages.log.fileSystemRealmMissingAttribute("algorithm", path, streamReader.getLocation().getLineNumber(), name);
+                    }
```

Comment:
This is necessary for BASE64 format only - when MCF format is used, algorithm is decoded from text.
Move this test into the condition.